docs/resources/client/models/modelloaders.md

```diff
@@ -0,0 +1,444 @@
+# Custom Model Loaders
+
+A model is simply a shape. It can be a cube, a collection of cubes, a collection of triangles, or any other geometrical shape (or collection of geometrical shape). For most contexts, it is not relevant how a model is defined, as everything will end up as a `BakedModel` in memory anyway. As such, NeoForge adds the ability to register custom model loaders that can transform any model you want into a `BakedModel` for the game to use.
+
+The entry point for a block model remains the model JSON file. However, you can specify a `loader` field in the root of the JSON that will swap out the default loader for your own loader. A custom model loader may ignore all fields the default loader requires.
+
+## Builtin Model Loaders
+
+Besides the default model loader, NeoForge offers a total of seven builtin loaders, each serving a different purpose.
+
+### Composite Model
+
+A composite model can be used to specify different model parts in the parent and only apply some of them in a child. This is best illustrated by an example. Consider the following parent model at `examplemod:example_composite_model`:
+
+```json5
+{
+  "loader": "neoforge:composite",
+  // Specify model parts.
+  "children": {
+    "part_1": {
+      "parent": "examplemod:some_model_1"
+    },
+    "part_2": {
+      "parent": "examplemod:some_model_2"
+    }
+  },
+  "visibility": {
+    // Disable part 2 by default.
+    "part_2": false
+  }
+}
+```
+
+Then, we can disable and enable individual parts in a child model of `examplemod:example_composite_model`:
+
+```json5
+{
+  "parent": "examplemod:example_composite_model",
+  // Override visibility. If a part is missing, it will use the parent model's visibility value.
+  "visibility": {
+    "part_1": false,
+    "part_2": true
+  }
+}
+```
+
+To [datagen][modeldatagen] this model, use the custom loader class `CompositeModelBuilder`.
+
+### Dynamic Fluid Container Model
+
+The dynamic fluid container model, also called dynamic bucket model after its most common use case, is used for items that represent a fluid container (such as a bucket or a tank) and want to show the fluid within the model. This only works if there is a fixed amount of fluids (e.g. only lava and powder snow) that can be used, use a [block entity renderer][ber] instead if the fluid is arbitrary.
```

Comment:
Doesn't this still have the base and cover which defines the container for the fluid model?